staging/src/k8s.io/apimachinery/pkg/runtime/codec_check.go

```diff
@@ -32,7 +33,17 @@ func CheckCodec(c Codec, internalType Object, externalTypes ...schema.GroupVersi
 		return fmt.Errorf("Internal type not encodable: %v", err)
 	}
 	for _, et := range externalTypes {
-		exBytes := []byte(fmt.Sprintf(`{"kind":"%v","apiVersion":"%v"}`, et.Kind, et.GroupVersion().String()))
+		info := struct {
+			Kind       string `json:"kind"`
+			ApiVersion string `json:"apiVersion"`
```

Comment:
this can be TypeMeta instead of an anonymous struct?
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L41

Reply:
done